### PR TITLE
feat(hack/build): release tooling

### PR DESCRIPTION
**Pull request by GeorgeMac**
This is clearing up some of the tooling we have around release management.

It removes the automatic submodule publishing. Since this is not quite the process we want or need to follow.

In provides a few new manual / local tools to add in versioning and tagging releases:
```sh
cd hack/build

mage release:latest [module]                   # latest without pre-releases (rcs)
mage release:latestrc [module]                 # latest including pre-releases (rcs)
mage release:next [module] [version-bump-spec] # next release based on version bump spec
mage release:tag [module] [version]            # tag module with release version

# version-bump-spec examples
# one of rc, patch, minor or major
# or chain with +
# e.g. minor+rc or patch+minor+major

# examples
mage release:latest go.flipt.io/flipt               # for latest flipt release version
mage release:next go.flipt.io/flipt/sdk/go minor+rc # for next sdk/go minor RC
mage release:tag go.flipt.io/flipt/sdk/go $(mage release:next minor+rc) # to tag next minor RC
```

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -91,7 +91,9 @@ jobs:
         env:
           TAG: ${{ github.ref_name }}
         run: |
-          if [[ $TAG == $(cat version.txt) ]]; then
+          cd hack/build
+
+          if [[ $TAG == $(mage release:latest go.flipt.io/flipt) ]]; then
```
Comment (markphelps) [inline .github/workflows/release.yml:96]: ♥️

File: hack/build/release/release.go [hunk 2]
```diff
@@ -2,116 +2,357 @@ package release
...
 	}
 
-	for _, submodule := range []string{
-		"errors",
-		"rpc/flipt",
-	} {
-		diff, err := base.WithExec([]string{"git", "diff", "--shortstat", lastTwo[0], lastTwo[1], "--", submodule}).Stdout(ctx)
-		if err != nil {
+	// go-git doesn't support half the functionality of git log
+	// It ends up brining along a lot of extra commits.
```
Comment (markphelps) [inline hack/build/release/release.go:146]:
```suggestion
	// It ends up bringing along a lot of extra commits.
```

File: hack/build/release/release.go [hunk 3]
```diff
@@ -2,116 +2,357 @@ package release
...
+	cmd := exec.Command("sh", "-c", logCmd)
+	out, err := cmd.CombinedOutput()
+	if err != nil {
+		return err
+	}
+
+	messages := strings.Split(strings.TrimSpace(string(out)), "\n")
+
+	return insertChangeLogEntryIntoFile(
+		path.Join(prefix, "CHANGELOG.md"),
```
Comment (markphelps) [inline hack/build/release/release.go:167]: could we use this to update the 'main' CHANGELOG for Flipt in the future?

Comment (GeorgeMac) [reply hack/build/release/release.go:167]: Indeed we can  🪄 :
```
cd hack/build
mage release:changelog go.flipt.io/flipt $(mage release:next go.flipt.io/flipt minor)
```

The output isn't perfect, but it certainly speeds up the process.

Comment (markphelps) [reply hack/build/release/release.go:167]: `mage release:changelog go.flipt.io/flipt $(mage release:next go.flipt.io/flipt minor) | gpt4 > CHANGELOG.md` 💰

**Pull request merged**